### ISSUE TITLE
Rewrite all throwing methods to return Result<Value, Error>

### ISSUE THREAD

**Issue by timburks** [googleapis/google-auth-library-swift#34]
Swift 5 added a standard [Result enum](https://github.com/apple/swift-evolution/blob/master/proposals/0235-add-result.md) 
that represents function results that can include errors. This is more concise and seems preferable to try/catch exception handling.

Should we update this library to use Result instead of throw?

+ @0xced

**Comment by 0xced** [comment]
Indeed, I already started to work on this. I'll submit a pull request when it's ready.

**Comment by dcow** [comment]
@0xced is this something you'd still submit if bumped or have you moved on to other things?

**Comment by 0xced** [comment]
I still have pending work on my [result](https://github.com/0xced/google-auth-library-swift/tree/result) branch that I have not even committed. It was almost two years ago, it may not even compile… But if you're interested, I can commit and push those changes so that someone else can take over.
